modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/QueryParser.java

```diff
@@ -553,6 +553,8 @@ else if (!prepared.isQuery()) {
                 GridCacheTwoStepQuery twoStepQry = null;
 
                 if (splitNeeded) {
+                    GridQueryOptimizer.pullOutSubQueries(selectStmt);
```

Comment:
GridQueryOptimizer - the name doesn't show its meaning. Is there any plans to do any other query optimization except subquery extraction? If yes - it should be named like GridSubueryJoinOptimizer

Reply:
Right now it does only one thing - rewrites subqueries. So ok, let's rename it